consensus/Cargo.toml

```diff
@@ -24,6 +25,7 @@ itertools = "0.10"
 log = "0.4"
 parking_lot = "0.12"
 rand = "0.8"
+thiserror = "1.0.38"
```

Comment:
We shouldn't be explicitly specifying the patch version.

Also, we don't use `anyhow`/`thiserror` anywhere else although we likely should. We were also trying to minimize the dependencies of ABFT, but these two are likely fine... So yeah, this second paragraph is only informational I guess, unless someone wants to oppose adding the deps.